ang/test/civicase/PanelQuery.spec.js

```diff
@@ -0,0 +1,512 @@
+/* eslint-env jasmine */
+(function ($, _) {
+  describe('panelQuery', function () {
+    var element, $compile, $q, $rootScope, $scope, crmApi, isolatedScope, mockedResults;
+    var NO_OF_RESULTS = 10;
+
+    beforeEach(module('civicase.templates', 'civicase', 'crmUtil'));
+
+    beforeEach(inject(function (_$compile_, _$q_, _$rootScope_, _crmApi_) {
+      $compile = _$compile_;
+      $q = _$q_;
+      $rootScope = _$rootScope_;
+      crmApi = _crmApi_;
+
+      $scope = $rootScope.$new();
+      mockedResults = _.times(NO_OF_RESULTS, function () {
+        return { id: _.random(1, 10) };
+      });
+
+      crmApi.and.returnValue($q.resolve({
+        get: { values: mockedResults },
+        count: NO_OF_RESULTS
+      }));
+    }));
+
+    describe('[query] attribute', function () {
+      beforeEach(function () {
+        $scope.queryData = { entity: 'Foo', params: { foo: 'foo' } };
+        compileDirective();
+      });
+
+      it('store its value in its scope', function () {
+        expect(isolatedScope.query).toBeDefined();
+        expect(isolatedScope.query).toEqual($scope.queryData);
+      });
```

Comment:
why is this test important? I don't think the query object is going to be used outside of the panel controller or link function.

Reply:
just making sure that the data is stored in the expected scope property